### PR TITLE
added python 3 support

### PR DESCRIPTION
**Pull request by de-code**
In the log when it still says that Apache Beam isn't yet fully supporting Python 3.
Although the [Python 2 and 3 support ticket](https://issues.apache.org/jira/browse/BEAM-1373) was resolved.

### CODE REVIEW

**Comment by de-code** [comment]
@giorgiosironi would it be easy to test over Python 2 and 3?

**Comment by giorgiosironi** [comment]
This project is container-based so if it provides both Python 2 and 3 both images should be built and tested. The Jenkinsfile may have more testing steps, but still a single `docker-compose build` should cover everything if both images are defined. They need not use separate Dockerfiles, as the base image could be a build argument; but it may help if the actual instructions are diverging.

**Comment by de-code** [comment]
> This project is container-based so if it provides both Python 2 and 3 both images should be built and tested. The Jenkinsfile may have more testing steps, but still a single `docker-compose build` should cover everything if both images are defined. They need not use separate Dockerfiles, as the base image could be a build argument; but it may help if the actual instructions are diverging.

I've done it in Docker as suggested. I believe the Docker builds are currently not run in parallel because there is no `--parallel` flag for the build command. Can the tests be run in parallel? e.g. it could then also show that they failed for PY2 but passed for PY3.

**Comment by giorgiosironi** [comment]
They can, but there's a race condition to solve in the `docker-compose` command.

**Comment by de-code** [comment]
> They can, but there's a race condition to solve in the `docker-compose` command.

I guess the `docker-compose down -v` also needs to be de-parallelised.

**Comment by giorgiosironi** [comment]
Back to you :handshake:

**Comment by de-code** [comment]
> Back to you

Thank you

File: Jenkinsfile [hunk 1]
```diff
@@ -43,7 +54,7 @@ elifeLibrary {
                 if (isNew) {
                     sh "IMAGE_TAG=${commit} " +
                         "docker-compose -f docker-compose.yml -f docker-compose.ci.yml run " +
-                        "sciencebeam-utils twine upload dist/*"
```
Comment (giorgiosironi) [inline Jenkinsfile]: after merge I'll change this to a tag-based build

Comment (de-code) [reply Jenkinsfile]: that would have been my next ququestion

**Pull request merged**